NewForce-Cohort-7/tabloidmvc-yellow-belly-yackers
# Ticket #8 - Edit a Post

# Description

**Completed ticket #8 [Edit a Post](https://trello.com/c/fw7cdfbq/8-8-edit-a-post)
Added functionality to the Edit button on Posts. 
## Type of change

# Testing Instructions

**GIT this branch:**

1. Clone this repository to your own machine using [git clone https://github.com/NewForce-Cohort-7/tabloidmvc-yellow-belly-yackers.git](https://github.com/NewForce-Cohort-7/tabloidmvc-yellow-belly-yackers/compare/git%20clone%20https:/github.com/NewForce-Cohort-7/tabloidmvc-yellow-belly-yackers.git) 
2. `git fetch -a` in the master branch to retrieve this branch named bh-editPost.
3.  checkout this branch using the command `git checkout bh-editPost`

**Create the database:**

1. Open Visual Studio.
2. At the start screen select "Open a Local Folder"
4. Navigate to your workspace directory, open the `tabloidmvc-yellow-belly-yackers` folder, then select the SQL folder
5. Run both of these SQL files [01_TabloidMVC_Create_DB.sql 02_TabloidMVC_Seed_Data.sql](https://github.com/NewForce-Cohort-7/tabloidmvc-yellow-belly-yackers/tree/main/SQL)

Now you need to add data to test the delete function for posts. 

> 1. In Visual Studio, Open the View menu and select SQL Server Object Explorer.
> 2. Expand the node for your SQL Server instance.
> 3. Expand "Databases"  
> 4. Right-click on the TabloidMVC database and choose New Query.
> 5. Run these SQL script to add a post to test the edit function.

> INSERT INTO [Post] ([Id], [Title], [Content], [ImageLocation], [CreateDateTime], [PublishDateTime], [IsApproved], [CategoryId], [UserProfileId])
VALUES (1, 'TEST DATA TO EDIT', 'GOING TO EDIT THIS.', 'https://gizmodiva.com/wp-content/uploads/2017/10/SCOTT-A-WOODWARD_1SW1943-1170x689.jpg',SYSDATETIME(), SYSDATETIME(), 1, 1, 1);
> INSERT INTO [Post] ([Id], [Title], [Content], [ImageLocation], [CreateDateTime], [PublishDateTime], [IsApproved], [CategoryId], [UserProfileId])
VALUES (1, 'OR TEST THIS DATA', 'EVEN MORE DATA TO TEST.', 'https://gizmodiva.com/wp-content/uploads/2017/10/SCOTT-A-WOODWARD_1SW1943-1170x689.jpg',SYSDATETIME(), SYSDATETIME(), 1, 1, 1);
> INSERT INTO [Post] ([Id], [Title], [Content], [ImageLocation], [CreateDateTime], [PublishDateTime], [IsApproved], [CategoryId], [UserProfileId])
VALUES (1, 'ADDING EVEN MORE DATA DATA DATA', 'GOING TO EDIT THIS DATA TIMES THREE.', 'https://gizmodiva.com/wp-content/uploads/2017/10/SCOTT-A-WOODWARD_1SW1943-1170x689.jpg',SYSDATETIME(), SYSDATETIME(), 1, 1, 1);

5. Now you have a posts to edit! 

**Test the new feature:**

1.	Click the Run button in visual studio to launch the site. 
2.	In the header, next to the Title (Tabloid) there should be a single link to Login, click this.
3.	For testing purposes, log in as admin@example.com, and click login.
4.	After successfully logging in, you should now see a link to MYPOSTS, click this.
5.	You will see the new data you added titled TEST DATA TO EDIT at the top of the list above the post titled OR TEST THIS DATA, followed by ADDING EVEN MORE DATA DATA DATA and C# is the Best Language.
6.	Click the button with the pencil to edit the post on any of the dummy data you added.
7.	This will take you to the edit form of the test data we added to edit. 
8.	Edit the Title of the post you chose to EDITED THIS TEST DATA WOO.
9.	Click the blue save button. This will take you back to the posts list and you can see that the title of the test data has been updated to EDITED THIS TEST DATA WOO. 



# Checklist:
